scripts/geo2fdn.py

```diff
@@ -0,0 +1,527 @@
+            alias = alias_prefix + ':' + entry.acc
+            bs.write(row, sheet_dict_bs['aliases'], alias)
+            bs.write(row, sheet_dict_bs['description'], entry.description)
+            if 'BiosampleCellCulture' in book.sheet_names():
+                bs.write(row, sheet_dict_bs['cell_culture_details'], alias + '-cellculture')
+            # bs.write(row, sheet_dict_bs['treatments'], entry.treatments)
+            bs.write(row, sheet_dict_bs['dbxrefs'], 'BioSample:' + entry.acc)
+            row += 1
+
+    if 'BiosampleCellCulture' in book.sheet_names():
+        sheet_dict_bcc = {}
+        bcc_sheets = book.sheet_by_name('BiosampleCellCulture').row_values(0)
+        for item in bcc_sheets:
+            sheet_dict_bcc[item] = bcc_sheets.index(item)
+        bcc = outbook.get_sheet('BiosampleCellCulture')
+        row = book.sheet_by_name('BiosampleCellCulture').nrows
+        print("Writing BiosampleCellCulture sheet...")
+        for entry in gds.biosamples:
+            bcc.write(row, sheet_dict_bcc['aliases'], alias_prefix + ':' + entry.acc + '-cellculture')
+            row += 1
+
+    file_dict = {}
+    if 'FileFastq' in book.sheet_names():
+        sheet_dict_fq = {}
+        fq_sheets = book.sheet_by_name('FileFastq').row_values(0)
+        for item in fq_sheets:
+            sheet_dict_fq[item] = fq_sheets.index(item)
+        fq = outbook.get_sheet('FileFastq')
+        row = book.sheet_by_name('FileFastq').nrows
+        print("Writing FileFastq sheet...")
+        for entry in gds.experiments:
+            file_dict[entry.geo] = []
+            for run in entry.runs:
+                if entry.layout.lower() == 'paired':
+                    fq1 = alias_prefix + ':' + run + '_1_fq'
+                    fq2 = alias_prefix + ':' + run + '_2_fq'
+                    file_dict[entry.geo] += [fq1, fq2]
+                    fq.write(row, sheet_dict_fq['aliases'], fq1)
+                    # fq.write(row, sheet_dict_fq['description'], entry.description)
+                    fq.write(row, sheet_dict_fq['*file_format'], 'fastq')
+                    fq.write(row, sheet_dict_fq['paired_end'], '1')
+                    fq.write(row, sheet_dict_fq['related_files.relationship_type'], 'paired with')
+                    fq.write(row, sheet_dict_fq['related_files.file'], fq2)
+                    fq.write(row, sheet_dict_fq['read_length'], entry.length)
+                    fq.write(row, sheet_dict_fq['instrument'], entry.instr)
+                    fq.write(row, sheet_dict_fq['dbxrefs'], 'SRA:' + run)
+                    fq.write(row + 1, sheet_dict_fq['aliases'], fq2)
+                    # fq.write(row + 1, sheet_dict_fq['description'], entry.description)
+                    fq.write(row + 1, sheet_dict_fq['*file_format'], 'fastq')
+                    fq.write(row + 1, sheet_dict_fq['paired_end'], '2')
+                    fq.write(row + 1, sheet_dict_fq['related_files.relationship_type'], 'paired with')
```

Comment:
It's better to only include one side of the paired_with relationship in the fastq sheet - the reciprocal relationship is generated automatically by the _update method in the types/file.py.